### PR TITLE
PLNSRVCE-700: use volumeClaimTemplate instead of static PVC

### PR DESCRIPTION
**Pull request by tkdchen**
Signed-off-by: Chenxiong Qi <cqi@redhat.com>

### CODE REVIEW

File: controllers/component_build_controller.go [hunk 1]
```diff
@@ -672,9 +667,19 @@ func GeneratePipelineRun(component appstudiov1alpha1.Component, bundle string, o
 			Params: params,
 			Workspaces: []tektonapi.WorkspaceBinding{
 				{
-					Name:                  "workspace",
-					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{ClaimName: "appstudio"},
-					SubPath:               pipelineName + "-{{revision}}",
+					Name: "workspace",
+					VolumeClaimTemplate: &corev1.PersistentVolumeClaim{
```
Comment (mmorhun) [inline controllers/component_build_controller.go:671]: Does it makes sense to have a function that generates the `VolumeClaimTemplate`? Then use the function in application-service and here, so we have consistency on how the volume is created.

Comment (tkdchen) [reply controllers/component_build_controller.go:671]: Good point :+1:

**Comment by tkdchen** [comment]
Update: this pr includes the changes made by Michal in PR #63.

File: go.mod [hunk 2]
```diff
@@ -18,15 +18,23 @@ require (
 )
 
 require (
-	github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0 // indirect
-	golang.org/x/tools v0.1.10-0.20220218145154-897bd77cd717 // indirect
+	code.gitea.io/sdk/gitea v0.14.0 // indirect
```
Comment (mmorhun) [inline go.mod:21]: Please group all indirect dependencies under the same require block

File: controllers/component_build_controller.go [hunk 3]
```diff
@@ -618,8 +612,7 @@ func GeneratePipelineRun(component appstudiov1alpha1.Component, bundle string, o
 	labels := map[string]string{
 		ComponentNameLabelName:                  component.Name,
 		"pipelines.appstudio.openshift.io/type": "build",
-		"appstudio.openshift.io/application":    component.Spec.Application,
-		"appstudio.openshift.io/component":      component.Name,
+		ApplicationNameLabelName:                component.Spec.Application,
```
Comment (mmorhun) [inline controllers/component_build_controller.go:615]: I'd prefer to have application and component labels one next to the other (not separated by another label).

**Review by mmorhun — APPROVED** [review]
LGTM if my remarks fixed

File: controllers/component_build_controller.go (hunk 3 above)
Comment (tkdchen) [reply controllers/component_build_controller.go]: Both are updated. PTAL.

**Review by Michkov — APPROVED** [review]

**Pull request merged**